ShipMonitor/ShipMonitor.cs

```diff
@@ -83,7 +82,7 @@ public UserControl ConfigurationTabItem()
 
         public void Save()
         {
-            uiSyncContext.Post(_ => writeShips(), null);
+            SynchronizationContext.Current.Post(_ => writeShips(), null);
```

Comment:
For future reference, this change is bad. You can only retreive the sync context from the main thread. Reinstating my code.